ui/pages/SummaryData/reducers.js

```diff
@@ -43,21 +43,32 @@ export const loadSuccessStory = successStoryTypes => (dispatch) => {
 
 export const loadSavedVariants = ({ tag, gene = '' }) => (dispatch, getState) => {
   // Do not load if already loaded
+  let tags = [SHOW_ALL]
   if (tag) {
-    if (getState().savedVariantTags[tag]) {
+    const { savedVariantTags } = getState()
+    tags = tag.split(';').filter(t => !savedVariantTags[t])
+    if (tags.length === 0) {
       return
     }
   } else if (!gene) {
     return
   }
 
   dispatch({ type: REQUEST_SAVED_VARIANTS })
-  new HttpRequestHelper(`/api/summary_data/saved_variants/${tag}`,
+  new HttpRequestHelper(`/api/summary_data/saved_variants/${tags.join(';')}`,
     (responseJson) => {
       if (tag && !gene) {
+        if (tags[0] === SHOW_ALL) {
+          tags = SUMMARY_PAGE_SAVED_VARIANT_TAGS
+        }
         dispatch({
           type: RECEIVE_SAVED_VARIANT_TAGS,
-          updates: { [tag]: true },
+          updates: tags.reduce((acc, t) => ({ ...acc, [t]: true }), {}),
+        })
+      } else if (gene) {
+        dispatch({
+          type: RECEIVE_SAVED_VARIANT_TAGS,
+          updates: SUMMARY_PAGE_SAVED_VARIANT_TAGS.reduce((acc, t) => ({ ...acc, [t]: false }), {}),
```

Comment:
why would we clear previously loaded data because we loaded more data?

Reply:
All the data loaded for the tags must be reloaded if the gene changes because the data differs for the genes with the same tags. I don't detect gene changes (I probably should). So clear tags loaded flag if the gene is not null to ensure correct results.

Reply:
but if we previously loaded like all the variants for a different tag, why would clear that? We should only clear for the tag(s) that were loaded